packages/miniapp-utils/src/helpers/proxying/utils.ts

```diff
@@ -0,0 +1,121 @@
+import jwt from 'jsonwebtoken'
+import proxyAddr from 'proxy-addr'
+import { z } from 'zod'
+
+import type { IncomingMessage } from 'http'
+
+type ProxyConfig = {
+    address: string | Array<string>
+    secret: string
+}
+
+type ProxyId = string
+
+export type KnownProxies = Record<ProxyId, ProxyConfig>
+
+export type TrustProxyFunction = (proxyAddr: string, idx: number) => boolean
+
+const _ipSchema = z.union([z.ipv4(), z.ipv6()])
+const _timeStampBasicRegexp = /^\d+$/
+
+const DEFAULT_PROXY_TIMEOUT_IN_MS = 5_000 // 5 sec to pass request is enough
```

Comment:
Could this be moved to ttl of the signature?

Reply:
It's duplicated in TTL signature as well. double check